### DISCUSSION TITLE
Use App check on react native app

### DISCUSSION THREAD

**Discussion by RyadhB2** [invertase/react-native-firebase#6698] [Q&A]
Hello,
I want to use Firebase app check to add a protection layer on the backend's rest APIs , so that only requests from my app are accepted . 
My firebase project has 2 apps, Android and iOS. I've read and followed every step on the docs [here](https://rnfirebase.io/app-check/usage)  and also [here](https://firebase.google.com/docs/app-check) ( to setup the android and iOS projects) 
My problem is even if I've followed all the steps and installed the sdk correctly, the documentation doesn't precise exactly how to use the app check sdk for react native, I've tried to follow the web + (Android || iOS) instructions but I always get errors obviously related to the code becasue the configuration isn't right ( because ther's no examples on the docs ) 
Here is a little code snippet of how I'm trying to activate app check on android then getting the token : 
`FB_appCheck.appCheck().activate('ignored', false).then((res) => {
      FB_appCheck.appCheck().getToken().then((res) => {
        console.log("generated token : ", res)
      })
    })`
I know that more steps have to be made for debugging using Debug Providers , but the main issue is that there aren't any documentation or example on how toi set everything up correctely on react native apps
here is the latest error I'me getting while trying to get the token:
![Screen Shot 2022-11-15 at 2 14 56 PM](https://user-images.githubusercontent.com/52285166/201928697-5ad55eed-fd1a-47db-afff-9c48d01a3f81.png)
[appCheck/token-error] java.net.UnknownHostException: Unable to resolve host "firebaseappcheck.googleapis.com": No address associated with hostname.

If anyone faced the same issue before and have an exaple of how to use appcheck on react native that would be a big help, thank you.

**Comment by mikehardy** [comment]
That's a networking error, there is something wrong with your local network. Here is what it looks like when I test on my local network:

```
mike@bistromath:~ % ping firebaseappcheck.googleapis.com
PING firebaseappcheck.googleapis.com (172.217.28.106) 56(84) bytes of data.
64 bytes from bog02s07-in-f10.1e100.net (172.217.28.106): icmp_seq=1 ttl=116 time=126 ms
64 bytes from gru06s09-in-f106.1e100.net (172.217.28.106): icmp_seq=2 ttl=116 time=126 ms
64 bytes from bog02s07-in-f10.1e100.net (172.217.28.106): icmp_seq=3 ttl=116 time=126 ms
^C
--- firebaseappcheck.googleapis.com ping statistics ---
4 packets transmitted, 3 received, 25% packet loss, time 3003ms
rtt min/avg/max/mdev = 125.866/126.018/126.256/0.170 ms
mike@bistromath:~ % 
```

And when I try it from the android emulator running on a test VM, it also works:

```
mike@osxvm:~ % adb shell
emulator64_x86_64_arm64:/ $ ping firebaesappcheck.googleapis.com                                                 
PING firebaesappcheck.googleapis.com (172.217.30.202) 56(84) bytes of data.
64 bytes from bog02s08-in-f10.1e100.net (172.217.30.202): icmp_seq=1 ttl=255 time=161 ms
64 bytes from bog02s08-in-f10.1e100.net (172.217.30.202): icmp_seq=2 ttl=255 time=73.3 ms
64 bytes from bog02s08-in-f10.1e100.net (172.217.30.202): icmp_seq=3 ttl=255 time=73.8 ms
^C
--- firebaesappcheck.googleapis.com ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2003ms
rtt min/avg/max/mdev = 73.305/102.792/161.268/41.349 ms
emulator64_x86_64_arm64:/ $ 
```

If you cannot pass traffic to the appcheck API server via your local network path, you won't be able to use AppCheck

**Reply by RyadhB2** [reply]
Thanks for the quick answer, but can you please provide me an example on how to set up app check correctely on my react native app?

**Reply by mikehardy** [reply]
You have mentioned in your original issue that the documentation is lacking but I do not think it actually is.
There just is not a lot to do.
If you were more specific https://stackoverflow.com/help/how-to-ask and provided an example of something that was not working as expected https://stackoverflow.com/help/minimal-reproducible-example perhaps I could offer more assistance.
